src/utils.jl

```diff
@@ -82,3 +82,6 @@ _maybewrapoffset(r::AbstractUnitRange, of, ::Base.OneTo) = no_offset_view(r)
 _maybewrapoffset(r::AbstractVector, of, ::Base.OneTo) = no_offset_view(r)
 _maybewrapoffset(r::AbstractUnitRange, of, ::Any) = IdOffsetRange(UnitRange(r), of)
 _maybewrapoffset(r::AbstractVector, of, axs) = OffsetArray(r .+ of, axs)
+
+_maybemap(::Type{T}, M::AbstractArray{T}) where {T} = M
+_maybemap(T, M::AbstractArray) = map(T, M)
```

Comment:
Mimic `MappedArrays.of_eltype`:

```suggestion
_of_eltype(::Type{T}, M::AbstractArray{T}) where {T} = M
_of_eltype(T, M::AbstractArray) = map(T, M)
```